control/control_loop.go

```diff
@@ -300,17 +300,19 @@ func (l *Loop) Stop() {
 
 // Pause sets l.running to false to pause the loop.
 func (l *Loop) Pause() {
-	for _, b := range l.pidBlocks {
-		if err := b.Reset(context.Background()); err != nil {
-			l.logger.Error(err)
-			return
-		}
-	}
 	l.running.Store(false)
 }
 
 // Resume sets l.running to true to resume the loop.
 func (l *Loop) Resume() {
+	if !l.running.Load() {
```

Comment:
the other fix potential fix is to reset the PID blocks when a user goes to resume, rather than when they pause. After testing this worked fine, just needed to call resume after the setpoints got updated.

Reply:
I tried resetting on resume which does work but I took it out because that means if you call set velocity and the base is moving, and then you call set velocity again with a new velocity, it will briefly pause. I think this is not the end of the world but I do see how it's not ideal

Reply:
oh this only resets if the previous state was paused to avoid that behavior.